DVCodingStyle.md

```diff
@@ -461,6 +462,125 @@ utility macros for even more common code sequences.
         If the macro is defined in a local perspective, it must be undefined at
         the end of the file, otherwise it will pollute the global namespace.
 
+#### Typical gotchas with macro usage:
+
+While it is better to avoid macros altogether, they do make our lives easy by
+shortening pieces of repeated code, in a manner that cannot be achieved with a
+function or a task. In general, the following describes the best practices to
+avoid common gotchas with macro usage.
+
+1.  Always wrap macro arguments **and** the output as applicable, within
+    parenthesis.
+
+    The following example illustrates how not wrapping the formal macro
+    arguments in parenthesis can cause unexpected behavior.
+    :-1:
+    ```systemverilog
+    `define AND(a, b) a && b
+
+    assign x = `AND(p || q, r) || s;
+    // Gotcha! wrong operator precedence! This results in: p || (q && r) || s which
+    // is different from what was originally intended: ((p || q) && r) || s.
+    ```
+    :+1:
+    ```systemverilog
+    `define AND(a, b) ((a) && (b))
+
+    assign x = `AND(p || q, r) || s;
+    // Correct: ((p || q) && (r)) || s
+    ```
+2.  Macro usage scope and avoiding namespace collisions.
+
+    *   If macros are defined for use in multiple sources, place them in a
+        separate SystemVerilog header file with appropriate `ifdef` guards.
+
+        This header file must not be compilation unit (provided as an input to
+        the tool for compilation). Instead, it should be `` `included`` in the
+        package, module or the interface that intends to invoke them.
```

Comment:
Need to rewrite this section - `` `included`` files get compiled, so the "must not be in a compilation unit" statement is invalid.

Reply:
Done.